src/sql/src/plan/statement/dml.rs

```diff
@@ -250,123 +244,131 @@ pub fn describe_explain(
+
+    Ok(StatementDesc::new(Some(relation_desc))
+        .with_params(describe_select(scx, SelectStatement { query, as_of: None })?.param_types))
+}
+
+pub fn plan_explain_plan(
     scx: &StatementContext,
-    ExplainStatement {
+    ExplainPlanStatement {
         stage,
         config_flags,
         format,
         no_errors,
         explainee,
-    }: ExplainStatement<Aug>,
+    }: ExplainPlanStatement<Aug>,
     params: &Params,
 ) -> Result<Plan, PlanError> {
-    let is_view = matches!(explainee, Explainee::View(_));
-    let (explainee, query) = match explainee {
-        Explainee::View(name) => {
-            let view = scx.get_item_by_resolved_name(&name)?;
-            let item_type = view.item_type();
-            // Return a more helpful error on `EXPLAIN [...] VIEW <materialized-view>`.
-            if item_type == CatalogItemType::MaterializedView {
-                return Err(PlanError::ExplainViewOnMaterializedView(
-                    name.full_name_str(),
-                ));
-            } else if item_type != CatalogItemType::View {
-                sql_bail!(
-                    "Expected {} to be a view, not a {}",
-                    name.full_name_str(),
-                    item_type
-                );
-            }
-            let parsed = crate::parse::parse(view.create_sql())
-                .expect("Sql for existing view should be valid sql");
-            let query = match parsed.into_last().ast {
-                Statement::CreateView(CreateViewStatement {
-                    definition: ViewDefinition { query, .. },
-                    ..
-                }) => query,
-                _ => panic!("Sql for existing view should parse as a view"),
-            };
-            let qcx = QueryContext::root(scx, QueryLifetime::OneShot);
-            (
-                mz_repr::explain::Explainee::Dataflow(view.id()),
-                names::resolve(qcx.scx.catalog, query)?.0,
-            )
+    let format = match format {
+        mz_sql_parser::ast::ExplainFormat::Text => ExplainFormat::Text,
+        mz_sql_parser::ast::ExplainFormat::Json => ExplainFormat::Json,
+        mz_sql_parser::ast::ExplainFormat::Dot => ExplainFormat::Dot,
+    };
+
+    let config = {
+        let config_flags = config_flags
+            .iter()
+            .map(|ident| ident.to_string().to_lowercase())
+            .collect::<BTreeSet<_>>();
+
+        let mut config = ExplainConfig::try_from(config_flags)?;
+
+        if config.filter_pushdown {
+            scx.require_feature_flag(&vars::ENABLE_MFP_PUSHDOWN_EXPLAIN)?;
+            // If filtering is disabled, explain plans should not include pushdown info.
+            config.filter_pushdown = scx.catalog.system_vars().persist_stats_filter_enabled();
+        }
+
+        config
+    };
+
+    let explainee = match explainee {
+        Explainee::View(_) => {
+            bail_never_supported!(
+                "EXPLAIN ... VIEW <view_name>",
```

Comment:
The problem with this suggestion is that if the view is indexed, then I'll just get a `ReadExistingIndex`, and no real plan. I'm not sure what would be a better suggestion. Maybe suggest calling EXPLAIN on an index if the view is indexed. Or maybe suggest calling EXPLAIN with the view's definition.

Reply:
> Maybe suggest calling EXPLAIN on an index if the view is indexed.

OK, I will add this suggestion when I add support for `EXPLAIN INDEX`.